log/labeler.go

```diff
@@ -0,0 +1,43 @@
+package log
+
+import (
+	"context"
+	"sync"
+
+	"go.uber.org/zap"
+)
+
+type LabelerContextKey string
+
+type Labeler struct {
+	mu     sync.Mutex
+	fields []zap.Field
+}
+
+// Add attributes to a Labeler.
+func (l *Labeler) Add(fields ...zap.Field) {
+	l.mu.Lock()
+	defer l.mu.Unlock()
+	l.fields = append(l.fields, fields...)
+}
+
+// Get returns a copy of the attributes added to the Labeler.
+func (l *Labeler) Get() []zap.Field {
+	l.mu.Lock()
+	defer l.mu.Unlock()
+	ret := make([]zap.Field, len(l.fields))
```

Comment:
How about making sure the fields are unique?

Reply:
I would stick to zap's behaviour in this case